src/openforms/submissions/utils.py

```diff
@@ -84,15 +86,50 @@ def send_confirmation_email(submission: Submission):
         logevent.confirmation_email_skip(submission)
         return
 
-    html_content = email_template.render(submission)
-    text_content = email_template.render(submission, {"rendering_text": True})
+    if (
+        submission.form.confirmation_email_option
+        == ConfirmationEmailOptions.form_specific_email
+    ):
+        try:
+            email_template = submission.form.confirmation_email_template
+            subject = email_template.subject
+            html_content = email_template.render(submission)
+            text_content = email_template.render(submission, {"rendering_text": True})
+        except (AttributeError, ObjectDoesNotExist):
+            logger.warning(
+                "No form specific confirmation email template for submission %d, "
+                "skipping the confirmation e-mail.",
+                submission.id,
+            )
+            logevent.confirmation_email_skip(submission)
+            return
+    elif (
+        submission.form.confirmation_email_option
+        == ConfirmationEmailOptions.global_email
+    ):
+        config = GlobalConfiguration.get_solo()
+        subject = Template(config.confirmation_email_subject).render(
+            Context({"form_name": submission.form.name})
+        )
+        html_content = config.render_confirmation_email_content(submission)
+        text_content = config.render_confirmation_email_content(
+            submission, {"rendering_text": True}
+        )
```

Comment:
As mentioned earlier - please extract this into a new function to obtain the subject/body template to use.